snippets/product-thumbnail.liquid

```diff
@@ -20,56 +21,40 @@
+  {%- else -%}
+    (min-width: {{ settings.page_width }}px) {{ settings.page_width | minus: 100 | times: media_width | round }}px, (min-width: 990px) calc({{ media_width | times: 100 }}vw - 10rem), (min-width: 750px) calc((100vw - 11.5rem) / 2), calc(100vw - 4rem)
+  {%- endif -%}
+{%- endcapture -%}
+
 <noscript>
   {%- if media.media_type == 'video' or media.media_type == 'external_video' -%}
     <span class="product__media-icon motion-reduce quick-add-hidden">{% render 'icon-play' %}</span>
     <div class="product__media media gradient global-media-settings" style="padding-top: {{ 1 | divided_by: media.preview_image.aspect_ratio | times: 100 }}%;">
-      <img
-        srcset="{% if media.preview_image.width >= 493 %}{{ media.preview_image | image_url: width: 493 }} 493w,{% endif %}
-          {% if media.preview_image.width >= 600 %}{{ media.preview_image | image_url: width: 600 }} 600w,{% endif %}
-          {% if media.preview_image.width >= 713 %}{{ media.preview_image | image_url: width: 713 }} 713w,{% endif %}
-          {% if media.preview_image.width >= 823 %}{{ media.preview_image | image_url: width: 823 }} 823w,{% endif %}
-          {% if media.preview_image.width >= 990 %}{{ media.preview_image | image_url: width: 990 }} 990w,{% endif %}
-          {% if media.preview_image.width >= 1100 %}{{ media.preview_image | image_url: width: 1100 }} 1100w,{% endif %}
-          {% if media.preview_image.width >= 1206 %}{{ media.preview_image | image_url: width: 1206 }} 1206w,{% endif %}
-          {% if media.preview_image.width >= 1346 %}{{ media.preview_image | image_url: width: 1346 }} 1346w,{% endif %}
-          {% if media.preview_image.width >= 1426 %}{{ media.preview_image | image_url: width: 1426 }} 1426w,{% endif %}
-          {% if media.preview_image.width >= 1646 %}{{ media.preview_image | image_url: width: 1646 }} 1646w,{% endif %}
-          {% if media.preview_image.width >= 1946 %}{{ media.preview_image | image_url: width: 1946 }} 1946w,{% endif %}
-          {{ media.preview_image | image_url }} {{ media.preview_image.width }}w"
-        src="{{ media | image_url: width: 1946 }}"
-        sizes="(min-width: {{ settings.page_width }}px) {{ settings.page_width | minus: 100 | times: media_width | round }}px, (min-width: 990px) calc({{ media_width | times: 100 }}vw - 10rem), (min-width: 750px) calc((100vw - 11.5rem) / 2), calc(100vw - 4rem)"
-        {% unless lazy_load == false %}loading="lazy"{% endunless %}
-        width="973"
-        height="{{ 973 | divided_by: media.preview_image.aspect_ratio | ceil }}"
-        alt="{{ media.preview_image.alt | escape }}"
-      >
+      {{ media.preview_image | image_url: width: 1946 | image_tag:
+        loading: lazy,
+        sizes: sizes,
+        widths: '493, 600, 713, 823, 990, 110, 1206, 1346, 1426, 1646, 1946',
+        alt: media.preview_image.alt | escape
+      }}
```

Comment:
Having the `width` and `height` attributes is good as they're meant to be there to help with layout shift. Having those sizes allows the browser to plan how much space is going to be taken. 
With the recently added flexibility of the product page, the number chosen was inaccurate as it doesn't account for the different sizes for the medias and wether or not it's 2 columns or stacked. 

And on top of that the image itself is positioned absolutely so we shouldn't need to have a `width` and `height` attribute I believe 🤔

Reply:
Thanks for sharing context about how setting both the `width` and `height` attributes were there to help with layout shift. And yeah, you're right about the number not being accurate initially (in this particular case: `973`).

I initially chose to do omit them based on ⬇️ but what you've shared is a much stronger argument that I completely agree with 👍🏼 

Referring to https://shopify.dev/api/liquid/filters#image_tag the `width` and `height` is automatically added based on the dimensions and aspect ratio from the image URL -- in this particular case that would be:

`{{ media.preview_image | image_url: width: 1946`

I was doing some comparative performance tests with and without the `width` and `height` attributes on the image tag and it turned out to be slightly faster without: https://github.com/Shopify/dawn/pull/1766#discussion_r943964982 -- these weren't thorough tests.